lib/liquidoc.rb

```diff
@@ -153,16 +175,58 @@ def validate_config_structure config
 # TODO More validation needed
 end
 
+def data_hashify data_var
+  # TODO make datasource config a class
+  if data_var.is_a?(String)
+    data = {}
+    data['file'] = data_var
+    data['ext'] = File.extname(data_var)
+  else # add ext to the hash
+    data = data_var
+    data['ext'] = File.extname(data['file'])
+  end
+  return data
+end
+
+def parse_regex data_file, pattern
+  records = []
+  pattern_re = /#{pattern}/
+  @logger.debug "Using regular expression #{pattern} to parse data file."
+  groups = pattern_re.names
+  begin
+    File.open(data_file, "r") do |file_proc|
+      file_proc.each_line do |row|
+        if row.length > 3 # ignore inherently invalid lines
+          matches = row.match(pattern_re)
+          if matches
+            row_h = {}
+            groups.each do |var| # loop over the named groups, adding their key & value to the row_h hash
+              row_h.merge! var => matches[var]
```

Comment:
nit: maybe include parentheses here since otherwise the expression looks a little confusing/ambiguous
`row_h.merge!(var => matches[var])`

Reply:
Ah cool, yeah that is helpful. Still barely learning syntax peculiarities of Ruby :-)